src/adapters/postgres/user-adapter.ts

```
@@ -2626,4 +2734,136 @@ export class PostgresUserService implements IServicelocator {
+            // Fetch the latest user from the database for upsert
+            const dbUser = await this.usersRepository.findOne({ where: { userId } });
+            if (!dbUser) return null;
+            let customFields = await this.fieldsService.getUserCustomFieldDetails(userId);
+            // Repeat the filtering for upsert callback
+            const formSubmissions = await this.fieldsValueRepository.manager.getRepository('FormSubmission').find({ where: { itemId: userId } });
+            const allFormFieldIds = new Set<string>();
+            for (const submission of formSubmissions) {
+              try {
+                const formRepo = this.fieldsValueRepository.manager.getRepository('Form');
+                const form = await formRepo.findOne({ where: { formid: submission.formId } });
+                const fieldsObj = form && form.fields ? (form.fields as any) : null;
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)